From ce903e0b402633992ff129ed97dd952a9b6fc801 Mon Sep 17 00:00:00 2001
From: Frank Wunderlich <frank-w@public-files.de>
Date: Thu, 9 Oct 2025 17:37:52 +0200
Subject: [PATCH 100/116] net: phy: mtk-2p5ge: extend priv for firmware loading

squashed:
net: phy: mtk-2p5ge: fix struct name for 2p5g phy priv
---
 drivers/net/phy/mediatek/mtk-2p5ge.c | 24 +++++++++++++++++-------
 drivers/net/phy/mediatek/mtk.h       |  1 +
 2 files changed, 18 insertions(+), 7 deletions(-)

diff --git a/drivers/net/phy/mediatek/mtk-2p5ge.c b/drivers/net/phy/mediatek/mtk-2p5ge.c
index e1ca312b1813..b4508bf7de67 100644
--- a/drivers/net/phy/mediatek/mtk-2p5ge.c
+++ b/drivers/net/phy/mediatek/mtk-2p5ge.c
@@ -116,7 +116,7 @@ enum {
 
 static int mt7987_2p5ge_phy_load_fw(struct phy_device *phydev)
 {
-	struct mtk_i2p5ge_phy_priv *priv = phydev->priv;
+	struct mtk_socphy_priv *priv = phydev->priv;
 	struct device *dev = &phydev->mdio.dev;
 	void __iomem *xbz_pcs_reg_base;
 	void __iomem *xbz_pma_rx_base;
@@ -637,6 +637,14 @@ static int mt798x_2p5ge_phy_probe(struct phy_device *phydev)
 	struct pinctrl *pinctrl;
 	int ret;
 
+	priv = devm_kzalloc(&phydev->mdio.dev, sizeof(struct mtk_socphy_priv),
+			    GFP_KERNEL);
+	if (!priv)
+		return -ENOMEM;
+
+	priv->fw_loaded = false;
+	phydev->priv = priv;
+
 	switch (phydev->drv->phy_id) {
 	case MTK_2P5GPHY_ID_MT7987:
 	case MTK_2P5GPHY_ID_MT7988:
@@ -690,22 +698,23 @@ static int mt798x_2p5ge_phy_probe(struct phy_device *phydev)
 	if (IS_ERR(pinctrl))
 		dev_err(&phydev->mdio.dev, "Fail to set LED pins!\n");
 
-	priv = devm_kzalloc(&phydev->mdio.dev, sizeof(struct mtk_socphy_priv),
-			    GFP_KERNEL);
-	if (!priv)
-		return -ENOMEM;
-	phydev->priv = priv;
-
 	mtk_phy_leds_state_init(phydev);
 
 	return 0;
 }
 
+static void mt798x_2p5ge_phy_remove(struct phy_device *phydev)
+{
+	struct mtk_i2p5ge_phy_priv *priv = phydev->priv;
+	kfree(priv);
+}
+
 static struct phy_driver mtk_2p5gephy_driver[] = {
 	{
 		PHY_ID_MATCH_MODEL(MTK_2P5GPHY_ID_MT7987),
 		.name = "MediaTek MT7987 2.5GbE PHY",
 		.probe = mt798x_2p5ge_phy_probe,
+		.remove = mt798x_2p5ge_phy_remove,
 		.config_init = mt798x_2p5ge_phy_config_init,
 		.config_aneg = mt798x_2p5ge_phy_config_aneg,
 		.get_features = mt798x_2p5ge_phy_get_features,
@@ -720,6 +729,7 @@ static struct phy_driver mtk_2p5gephy_driver[] = {
 		PHY_ID_MATCH_MODEL(MTK_2P5GPHY_ID_MT7988),
 		.name = "MediaTek MT7988 2.5GbE PHY",
 		.probe = mt798x_2p5ge_phy_probe,
+		.remove = mt798x_2p5ge_phy_remove,
 		.config_init = mt798x_2p5ge_phy_config_init,
 		.config_aneg = mt798x_2p5ge_phy_config_aneg,
 		.get_features = mt798x_2p5ge_phy_get_features,
diff --git a/drivers/net/phy/mediatek/mtk.h b/drivers/net/phy/mediatek/mtk.h
index 320f76ffa81f..5b27df8622b8 100644
--- a/drivers/net/phy/mediatek/mtk.h
+++ b/drivers/net/phy/mediatek/mtk.h
@@ -70,6 +70,7 @@
 
 struct mtk_socphy_priv {
 	unsigned long		led_state;
+	bool			fw_loaded;
 };
 
 void __mtk_tr_modify(struct phy_device *phydev, u8 ch_addr, u8 node_addr,
-- 
2.39.5

